File: microtime/microtime.py
# ATOMIC_FUNCTION:  microtime ()
# TAGS: time
# LANG:  python
# FUNCTION: microtime ()
import time
start_val = time.time()
time.sleep(0.1)
time_end = time.time()
end_val = time_end - start_val

# TEST
if(end_val > 0 and end_val < 1):
    print("PASS")
else:
    print("FAIL")

File: microtime/microtime.java
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class microtime {
    // ATOMIC_FUNCTION:  microtime()
    // TAGS: time
    // LANG:  java
    // FUNCTION:  microtime()
    public static long microtime(){
        return System.currentTimeMillis();
    }
    public static void main(String[] args){
        long start_val=microtime();
        try {
            Thread.sleep(1);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        long end_time=microtime();
        long end_val = end_time-start_val;


        // TEST
        if(end_val>0 && end_val<10) {
            System.out.println("PASS");
        } else {
            System.out.println("FAIL");
        }
    }
}
